components/AssetFormBuy/index.tsx

```diff
@@ -61,6 +63,9 @@ export const AssetFormBuy = ({
     setPurchasingError('');
     setIsBalanceInsufficient(false);
     setSaleId(id);
+    setAssetId(
+      dropdownAssets.filter((asset) => asset.saleId === id)[0].assetId
```

Comment:
In this situation, it would be better to use .find:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

That way you don't have to do the `[0]` and it exits out of the loop once it has been found.